CONTRIBUTING.md

```diff
@@ -10,6 +10,16 @@ This repository is HASH's public monorepo. It contains [several different](READM
 
 To ascertain the license and contributor policy for a given project, check out the `LICENSE.md` and `CONTRIBUTOR.md` files in its root.
 
+## Common contribution processes
+
+These apply across all projects:
+
+- Before undertaking any significant work, please share your proposal with us. You can do so as a [discussion](https://github.com/hashintel/hash/discussions) if it's a feature proposal, or an [issue](https://github.com/hashintel/hash/issues) if it's a bug you intend to fix. If you're unsure, you can always chat to us on [Discord](https://hash.ai/discord) first. If it's a typo or docs change, don't worry about this step.
```

Comment:
Do we perhaps want to give a nice reason to this?
Something like 
> We don't want you to waste your time and it's possible the feature conflicts with planned work, or is already underway. Because of this, significant or large pieces of work should be discussed first, you can do this by sharing a proposal on discussions, issues, etc.

Reply:
https://github.com/hashintel/hash/pull/345/commits/62fa2c59bb1c45e3f8e4e9ede6ed0489db70ef97